docs/features/incremental-generators.cookbook.md

```diff
@@ -636,7 +636,248 @@ TODO: https://github.com/dotnet/roslyn/issues/72149
+                    return new InterfacePropertyModel(type, interfaceProperty.Name, interfaceProperty.SetMethod is not null);
+                })
+                .ToArray();
+
+            string containingNamespace = interfaceSymbol.ContainingNamespace.ToString();
+
+            ret.Add(new InterfaceModel(interfaceName, containingNamespace, interfaceUsings, propertyModels));
+        }
+
+        return [.. ret];
+    }
+    private static string[] GetUsings(SyntaxNode root)
+    {
+        return root.DescendantNodes()
+                .OfType<UsingDirectiveSyntax>()
+                .Select(usingDirective =>
+                {
+                    //get string "System" from string "using System;"
+                    string usingName = usingDirective.ToString().Split(' ').Last().TrimEnd(';');
+                    return usingName;
+                })
+                .ToArray();
+    }
+    private static string[] GetInterfacesNames(SyntaxList<AttributeListSyntax> attributeLists)
+    {
+        foreach (AttributeListSyntax attributeList in attributeLists)
+        {
+            foreach (AttributeSyntax attribute in attributeList.Attributes)
+            {
+                if (!attribute.Name.ToString().Equals(AutoImplementAttributeName) || attribute.ArgumentList is null)
+                    continue;
+
+                return attribute.ArgumentList.Arguments
+                    .Select(x => GetInterfaceName(x.Expression.ToString()))
+                    .ToArray();
+            }
+        }
+
+        return [];
+    }
+    private static string GetInterfaceName(string expression)
+    {
+        // Empty String
+        if (string.IsNullOrWhiteSpace(expression))
+            return expression;
+
+        // Interface Name got by nameof()
+        if (expression.StartsWith(NameOfString))
+            return expression.Substring(NameOfString.Length, expression.Length - NameOfString.Length - 1);
+
+        //Trimming ""
+        string ret = expression.Trim('"');
+        return ret;
+    }
+    private static INamedTypeSymbol? GetInterfaceSymbol(Compilation compilation, string[] nameSpaces, string implementedInterfaceName)
+    {
+        // Try Get without NameSpace if interface was already Fully Qualified
+        INamedTypeSymbol? interfaceSymbol = compilation.GetTypeByMetadataName(implementedInterfaceName);
+        if (interfaceSymbol is not null)
+            return interfaceSymbol;
+
+        foreach (string nameSpace in nameSpaces)
+        {
+            interfaceSymbol = compilation.GetTypeByMetadataName($"{nameSpace}.{implementedInterfaceName}");
+            if (interfaceSymbol is not null)
+                return interfaceSymbol;
+        }
+
+        return null;
+    }
+
+    private record Model(string ClassName, string ClassNameSpace, InterfaceModel[] Interfaces);
+    private record InterfaceModel(string Name, string NameSpace, string[] Usings, InterfacePropertyModel[] Properties);
```

Comment:
Use an equatable list implementation, rather than arrays.

Reply:
I'm really grateful for all your suggestions
By 'equatable list implementation' do you mean a custom implementation like 'public class EquatableList<T> : List<T>, IEquatable<EquatableList<T>>' or would you prefer another solution?

Reply:
That would work. Here's what the cookbook says further up:

> Be careful of collection types in your models. Most built-in collection types in .NET do not do value equality by default. Arrays, ImmutableArray<T> and List<T>, for example, use reference equality, not value equality. We suggest that most generator authors use a wrapper type around arrays to augment them with value-based equality.